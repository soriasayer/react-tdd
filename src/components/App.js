import React, { Component } from 'react'
import { Button } from 'react-bootstrap'

class App extends Component{

  state = {
    gifts: []
  }

  addGift = () => {
    const {gifts} = this.state

    const ids = gifts.map(gift => gift.id)
    const max_id = ids.length > 0 ? Math.max(...ids) : 0
    gifts.push({id: max_id+1})

    this.setState({gifts})
  }

  render() {
    return (
      <div>
        <h2>Gift Giver</h2>
        <div className='gitft-list'>
        {this.state.gifts.map(gift => {
          return (
            <div key={gift.id}></div>
          )
        })}
        </div>
        <Button className='btn-add' onClick={this.addGift}>Add Gift</Button>
      </div>
    )
  }
}

export default App
