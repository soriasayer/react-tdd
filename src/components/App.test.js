import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import App from './App';

configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe('App', () => {
  const app = shallow(<App/>)

  it('render correctly', () => {
    expect(app).toMatchSnapshot()
  })

  it('initialize the `state` with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
  })

  it('adds a new gift to the `state` when clicking the `add-gift` button', ()=> {
    app.find('.btn-add').simulate('click')
    expect(app.state().gifts).toEqual([{id: 1}])
  })

  it('add a new gift to the rendered list when clicking on the `add gift` button', () => {
    app.find('.btn-add').simulate('click')
    expect(app.find('.gift-list').children().length).toEqual(0)
  })
})
