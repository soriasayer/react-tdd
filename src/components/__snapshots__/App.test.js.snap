// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly 1`] = `ShallowWrapper {}`;
